Dockerfile

```diff
@@ -0,0 +1,23 @@
+FROM node:12.2-alpine as prod
+WORKDIR /app/src
+
+# handle dependencies
+COPY package.json ./
+COPY yarn.lock ./
+RUN yarn
+
+# copy over source code
+COPY .eslintrc.json ./
+COPY test ./test
+COPY *.js ./
+COPY features ./features
+COPY plugins ./plugins
+COPY sample-script ./sample-script
+
+# run tests
+RUN yarn test
+
+ENV PORT 3000
+EXPOSE ${PORT}
+ENTRYPOINT ["yarn", "start"]
+
```

Comment:
we should be using a multistage build to keep the final image smaller.  the production image doesn't need to have the test files, or the `devDependencies`.  in the final image, you can run `yarn --production` to only install production dependencies.